### PR TITLE
[SPARK-29751][ML] Scalers use Summarizer instead of MultivariateOnlineSummarizer

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
use `ml.Summarizer` instead of `mllib.MultivariateOnlineSummarizer`

### Why are the changes needed?
1, I found that using `ml.Summarizer` is faster than current impl;
2, `mllib.MultivariateOnlineSummarizer` maintain all arrays, while `ml.Summarizer` only maintain necessary arrays
3, using `ml.Summarizer` will avoid vector conversions to `mlllib.Vector`

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
existing testsuites

### CODE REVIEW

**Comment by zhengruifeng** [comment]
test code
```scala
import org.apache.spark.ml.feature._

scala> var df = spark.read.format("libsvm").load("/data1/Datasets/a9a/a9a")
19/11/05 13:47:02 WARN LibSVMFileFormat: 'numFeatures' option not specified, determining the number of features by going though the input. If you know the number in advance, please specify it via 'numFeatures' option to avoid the extra scan.
df: org.apache.spark.sql.DataFrame = [label: double, features: vector]          

scala> df.persist()
res0: org.apache.spark.sql.Dataset[org.apache.spark.sql.Row] = [label: double, features: vector]


scala> df.count
res1: Long = 32561

scala> (0 until 8).foreach(_ => df = df.union(df))

scala> df.count
res3: Long = 8335616 

val durations1 = (0 until 50).map{i => val tic = System.currentTimeMillis; val scaler = new MaxAbsScaler().setInputCol("features"); val model = scaler.fit(df); val toc = System.currentTimeMillis; toc - tic}

durations1.takeRight(30).sum.toDouble / 30



val durations2 = (0 until 50).map{i => val tic = System.currentTimeMillis; val scaler = new MinMaxScaler().setInputCol("features"); val model = scaler.fit(df); val toc = System.currentTimeMillis; toc - tic}

durations2.takeRight(30).sum.toDouble / 30



val durations3 = (0 until 50).map{i => val tic = System.currentTimeMillis; val scaler = new StandardScaler().setInputCol("features"); val model = scaler.fit(df); val toc = System.currentTimeMillis; toc - tic}

durations3.takeRight(30).sum.toDouble / 30
```

Results:

|MaxAbsScaler(Old)| MinMaxScaler(Old) | StandardScaler(Old) | MaxAbsScaler(New) | MinMaxScaler(New) | StandardScaler(New) |
|------|----------|------------|----------|------------|----------|
|6768.1|6875.2|6899.9|5862.1|5880.3|5889.7|

**Comment by zhengruifeng** [comment]
IIRC, when `Summarizer` was created, I had done similar tests, and at that time `Summarizer` was much slower than `MultivariateOnlineSummarizer`.
I am surprised by the results and will look for whether it works in other places.

**Comment by zhengruifeng** [comment]
friendly ping @srowen @WeichenXu123

**Review by srowen — COMMENTED** [review]
Also looks OK as a refactoring and optimization - if it's faster.

File: mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala [hunk 1]
```diff
@@ -69,14 +68,13 @@ class MaxAbsScaler @Since("2.0.0") (@Since("2.0.0") override val uid: String)
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): MaxAbsScalerModel = {
     transformSchema(dataset.schema, logging = true)
-    val input = dataset.select($(inputCol)).rdd.map {
-      case Row(v: Vector) => OldVectors.fromML(v)
-    }
-    val summary = Statistics.colStats(input)
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala]: After this PR, `Statistics.colStats` is no longer directly used in the `.ml` side.
It is still used in `mllib.PCA`, which is the impl of `ml.PCA`

**Comment by zhengruifeng** [comment]
Thanks @srowen for reviewing!

**Pull request closed without merging**